### PR TITLE
Get profile picture and user details from Microsoft Graph API

### LINKED ISSUE #332: Change to user avatar when logged in
image

At the same time, show real name instead of email-address?

### PR DESCRIPTION
**Pull request by rubenthoms**
Fetching profile picture and user details from Microsoft Graph API. Taking in use `asyncio` backend for simultaneous get requests.

Closes #332.

### CODE REVIEW

File: backend/src/services/utils/authenticated_user.py [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
         self._username = username
-        self._sumo_access_token = sumo_access_token
-        self._smda_access_token = smda_access_token
-        self._pdm_access_token = pdm_access_token
-        self._ssdl_access_token = ssdl_access_token
+        self._graph_access_token = access_tokens.get("graph_access_token", None)
+        self._sumo_access_token = access_tokens.get("sumo_access_token", None)
+        self._smda_access_token = access_tokens.get("smda_access_token", None)
+        self._pdm_access_token = access_tokens.get("pdm_access_token", None)
+        self._ssdl_access_token = access_tokens.get("ssdl_access_token", None)
```
Comment (anders-kiaer) [inline backend/src/services/utils/authenticated_user.py:27]:
```suggestion
        self._graph_access_token = access_tokens.get("graph_access_token")
        self._sumo_access_token = access_tokens.get("sumo_access_token")
        self._smda_access_token = access_tokens.get("smda_access_token")
        self._pdm_access_token = access_tokens.get("pdm_access_token")
        self._ssdl_access_token = access_tokens.get("ssdl_access_token")
```
I think `None` is default value if key does not exist?

File: backend/src/backend/primary/routers/general.py [hunk 2]
```diff
@@ -47,10 +56,27 @@ async def logged_in_user(request: Request) -> UserInfo:
...
+            avatar_b64str_future = asyncio.create_task(graph_api_access.get_user_profile_photo())
+            graph_user_info_future = asyncio.create_task(graph_api_access.get_user_info())
+
+            avatar_b64str = await avatar_b64str_future
+            graph_user_info = await graph_user_info_future
+
+            user_info.avatar_b64str = avatar_b64str
+            if graph_user_info is not None:
+                user_info.display_name = graph_user_info.get("displayName", None)
+        except Exception as e:
```
Comment (anders-kiaer) [inline backend/src/backend/primary/routers/general.py:77]: Do we know which exceptions we are expecting here? Currently we catch all.

**Review by anders-kiaer — APPROVED** [review]
Two small comments, otherwise LGTM! 🚀🙂

**Review by HansKallekleiv — APPROVED** [review]
Nice getting `httpx` in 👍 
We should do the same for the other http requests we have.
SMDA: https://github.com/equinor/webviz/issues/346
Upcoming PR for VDS: https://github.com/equinor/webviz/pull/319

**Pull request merged**